forge/routes/api/team.js

```diff
@@ -22,8 +22,27 @@ module.exports = async function (app) {
         if (request.params.teamId !== undefined) {
             if (request.params.teamId) {
                 try {
-                    // if User is not defined, check to see if request.context.config.allowAnonymous is set
-                    if (request.session?.User === undefined && request.context.config?.allowAnonymous === true) {
+                    if (!request.session.User) {
+                        // If request.session.User is not defined, this request is being
+                        // made with an access token. If it is a project access token,
+                        // ensure that project is in this team
+                        if (request.session.ownerType === 'project') {
+                            // Want this to be as small a query as possible. Sequelize
+                            // doesn't make it easy to just get `TeamId` without doing
+                            // a join on Team table.
+                            const project = await app.db.models.Project.findOne({
+                                where: { id: request.session.ownerId },
+                                include: {
+                                    model: app.db.models.Team,
+                                    attributes: ['hashid', 'id']
+                                }
+                            })
+                            // Ensure the token's project is in the team being accessed
+                            if (project && project.Team.hashid === request.params.teamId) {
+                                return
+                            }
+                        }
+                        reply.code(404).type('text/html').send('Not Found')
```

Comment:
Nick, the original PR had `if (request.session?.User === undefined && request.context.config?.allowAnonymous === true) {` as there were instances while developing and restarting & jiggling about, the `session` object was not present. 

This ended up in a hard to escape loop.  I didnt manage to re-create this reliably and assumed this was due to the dev work BUT I suspect it was due to log debug periods and session timeout?

Reply:
With `allowAnonymous` set to `true` then yes, it would be possible for `request.session` to be undefined because `allowAnonymous` means allow requests that don't have a valid session or token.

That's why I've removed its use here.

Reply:
ah, makes sense. Thanks for clarifying.